### ISSUE TITLE
Document how to apply patch from mercurial to git

### ISSUE THREAD

**Issue by Mariatta** [python/devguide#193]
How I do it: 

1. In the CPython mercurial repo, re-create the patch using --git parameter: 
    hg diff --git > issueNNNN-git.patch

2. In the CPython git repo, create and switch to a new branch. Apply the patch into CPython git repository:
    git apply /path/to/issueNNNN-git.patch

3. Commit, push, and open the PR.

**Comment by Mariatta** [comment]
I think this can go in committing.rst

**Comment by DimitrisJim** [comment]
Wouldn't this be better in `gitdevs.rst` or `gitbootcamp.rst`?

**Comment by Mariatta** [comment]
gitdevs.rst (Mercurial for git developers)  might go away, as proposed in https://github.com/python/devguide/issues/120

About adding it to gitbootcamp.rst, it's fine by me :)

**Comment by brettcannon** [comment]
Does `patch -p1 < path/to/patch.diff` not work?

**Comment by Mariatta** [comment]
If it works, let's use `patch -p1 < path/to/patch.diff`.
I'm traveling today and on my phone right now so I can't try it myself 😝

**Comment by ezio-melotti** [comment]
Several patches on the tracker have already been created with `--git`, so for those, the 1st step is unnecessary.  `git apply` might also work with non-`--git` patches, but I haven't tested.
`patch` is generally not available on Windows.

**Comment by terryjreedy** [comment]
I believe git patches do not work with Rietveld, so people have often been asked to resubmit.  In any case, the top of the file indicates format.
Program 'patch' does not come with Windows; don't know if it exists.

**Comment by Mariatta** [comment]
Alright just got around trying out `patch`. It works on my Macbook.

If we want to document `patch` and mentioning it works for Mac & Unix only, the steps would be
1. In the CPython git repo, create and switch to a branch
2. `patch -p1 < /path/to/issueNNNN.patch`
3. commit, push and open the PR.

**Comment by kirbyfan64** [comment]
FWIW there's also Mozilla's [hg-patch-to-git-patch](https://github.com/mozilla/moz-git-tools/blob/master/hg-patch-to-git-patch), with which the entire workflow can be shortened to:

```
curl patch_url | hg-patch-to-git-patch | git am
```

**Comment by matrixise** [comment]
+1 for `git apply`.
in the past, I used `patch` with the patches from `hg.python.org` when I wanted to patch my working directory for a test and sometimes, I got an error.
Since I use `git apply`, it's really rare when I have an issue.

About `hg-patch-to-git-patch` maybe we could use it if we create a copy in the `tools` directory of CPython just for the maintenance, because if Mozilla stops the dev of this small script, we will discuss for a new tool, new way to do that.
